### ISSUE TITLE
Instead of deleting expired tokens: allow to reenable them

### ISSUE THREAD

**Issue by atar-axis** [anx-ckreuzberger/django-rest-passwordreset#52]
It would be pretty nice to offer the possibility to re-enable deactivated tokens instead of deleting them.

My use case:
Initially all my are going to get random passwords, so they cannot log in.
But I would like to give them the token as a "registration token", printed on a paper.
The users are asked to login and change their password (i.e. "to activate their account") within 24 hours, but I expect that some will not manage to change it in the given time.

Instead of printing a new activation token, it would be nice to just "re-enable" the old one for another 24 hours.

Is this something which is a lot of work? I am relatively new to Django so I fear I cannot do it myself, but I would be glad to be of any help you need.

**Comment by anx-ckreuzberger** [comment]
I don't think that this is a use-case that should be implemented within django-rest-passwordreset.

Re-enabling tokens is also sub-optimal, I would recommend to change the expiry time based on token type (e.g., you could have two token types, one for `registration` and one for `password_reset`).

As I'm time-constraint on supporting this library I would like to keep the footprint as small as possible, and not introduce new features.

I'll keep this issue open for others to comment on this, and maybe someone will create a fork that supports this behaviour.

**Comment by calhouncole** [comment]
@anx-ckreuzberger Are there any examples as to how one can change the expiry time based on token type (i.e. one for `registration` and one for `password_reset`) as you mention above?

It would be extremely useful if we could easily use this library's methodology for both use cases.

**Comment by nezhar** [comment]
Re-enabling expired tokens resurrects a secret that may have already leaked (printed on cards, sent via email, written to logs). Expiry should be irreversible. The correct fix for a missed/expired token is to issue a new one, not re-enable the old one.

For the registration-card use case, use a separate token type with its own TTL
